docs/_future/deploy_your_app.md

```diff
@@ -7,10 +7,256 @@ layout: tutorial
+
+With these set up, you can use the `slack run` command to make sure your app starts successfully and appropriately responds to Triggers from your machine.
+
+Since we're deploying to Heroku, having a Heroku account will also be useful. If you don't have one, [create one here](https://signup.heroku.com/).
+
+In order to manage, deploy, and debug your app on Heroku, you'll also need to [install the Heroku CLI](https://devcenter.heroku.com/articles/getting-started-with-nodejs#set-up) onto your machine and authenticate using `heroku login`.
+
+And lastly, to manage the version of your project being deployed, you'll need to [install Git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git) and initialize your app's repository using `git init` if you have not already done so.
+
+---
+
+### Create a Heroku app {#create-a-heroku-app}
+
+With a Bolt app ready to go and Heroku tools installed, you're now ready to start deploying your Bolt app on Heroku!
+
+#### Create a new app
+
+From the command line, go to your project directory and start by creating a Heroku app with a unique name:
+
+```sh
+$ heroku create your-heroku-app
+```
+
+or have some fun with a random name:
+
+```sh
+$ heroku create
+# Creating app... done, ⬢ aqueous-escarpment-85734
+# https://aqueous-escarpment-85734.herokuapp.com/ | https://git.heroku.com/aqueous-escarpment-85734.git
+```
+
+After your Heroku app is created, you'll be given app-specific information for use in later steps. From the example above:
+
+* App name: `aqueous-escarpment-85734`
+* Web address: `https://aqueous-escarpment-85734.herokuapp.com/`
+* Git remote: `https://git.heroku.com/aqueous-escarpment-85734.git`
+
+#### Add environment variables
+
+The tokens being used for your app (and any other environment variables) can be added to your deployed environment using the following commands:
+
+```sh
+$ heroku config:set SLACK_APP_TOKEN=xapp-your-app-level-token
+$ heroku config:set SLACK_BOT_TOKEN=xoxb-your-bot-token
+```
+
+These tokens can be collected from [your App Config page](https://api.slack.com/apps) if ever misplaced. The app-level token can be found on the "Basic Information" page, while the bot-level token can be found under "OAuth & Permissions".
+
+Note that these tokens and variables don't necessarily have to match those used in development! For instance, you may want to use tokens for a different app or a more verbose logging output in production. This is where you would set that.
+
+### Prepare your app for Heroku {#prepare-your-app}
+
+For a successful deployment, you'll need to set up a few additional things in your app.
+
+#### Add a Procfile {#procfile}
+
+To specify the start command of your app for Heroku, [a special file called `Procfile`](https://devcenter.heroku.com/articles/procfile) must be created in your app's root directory.
```

Comment:
Is this the root directory or just wherever the `app.js` file lives (which is typically the root?)

Reply:
This is the root directory of the project, where the `.git` folder also lives. Do you have any suggestions for how to make this more clear?

Reply:
I think root directory works in that context! 👍 i was wondering specifically because the `Procfile` contents reference an `app.js` that looks like it should be in the same directory, but I think this is already pretty clear and straightforward 🙌